src/interfaces/IHookMetadata.sol

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {IEIP712_v4} from "./IEIP712_v4.sol";
+import {IEIP7512} from "./IEIP7512.sol";
+
+/**
+ * @title IHookMetadata
+ * @notice This interface is designed so that external indexing services can discover
+ *         and display essential information about a Uniswap v4 hook. It extends the
+ *         on-chain audit representation outlined in EIP‑7512, thereby allowing the hook
+ *         to store and provide signed audit summaries for third-party verification.
+ *
+ * ----------------------------------------------------------------------------
+ * HOOK METADATA FLOW
+ *
+ * 1. Required Metadata
+ *    - Every hook must implement: name(), repository(), logoURI(), websiteURI(),
+ *      description(), version().
+ *    - These fields are effectively immutable and indexed at deployment time.
+ *
+ * 2. Audit Summaries
+ *    - The function auditSummaries(auditId) returns a completed audit summary.
+ *    - Audits can be appended over time, and each newly added audit summary must
+ *      emit the AuditSummaryRegistered event for indexers.
+ *
+ * 3. EIP-712 Domain Information
+ *    - The contract must provide an EIP-712 DOMAIN_SEPARATOR to allow auditors
```

Comment:
Is there a standard format you'd want to enforce in the DOMAIN_SEPARATOR? Could give a recommendation for which fields should/must be specified. ie Im guessing the verifyingContract is the hook but maybe worth specifying?

Reply:
We created a template for IHookMetadata usage - https://github.com/uniswapfoundation/v4-template/pull/58, used https://github.com/uniswap/v4-periphery/blob/ea2bf2e1ba6863bb809fc2ff791744f308c4a26d/src/base/EIP712_v4.sol there